src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -618,21 +618,45 @@ await foreach (ConfigurationSetting setting in client.GetConfigurationSettingsAs
                 }).ConfigureAwait(false);
             }
 
-                foreach (var loadOption in _options.KeyValueSelectors)
+            foreach (KeyValueSelector loadOption in _options.KeyValueSelectors)
+            {
+                IAsyncEnumerable<ConfigurationSetting> settingsEnumerable;
+
+                if (string.IsNullOrEmpty(loadOption.SnapshotName))
+                {
+                    settingsEnumerable = client.GetConfigurationSettingsAsync(
+                        new SettingSelector
+                        {
+                            KeyFilter = loadOption.KeyFilter,
+                            LabelFilter = loadOption.LabelFilter
+                        },
+                        cancellationToken);
+                }
+                else
                 {
-                    var selector = new SettingSelector
+                    ConfigurationSettingsSnapshot snapshot = await client.GetSnapshotAsync(loadOption.SnapshotName);
+
+                    if (snapshot != null)
                     {
-                        KeyFilter = loadOption.KeyFilter,
-                        LabelFilter = loadOption.LabelFilter
-                    };
+                        if (snapshot.CompositionType == null || !snapshot.CompositionType.ToString().Equals("key"))

```

Comment:
how can snapshot composition type be numb. Isn't it an enum?

Reply:
Okay, it's nullable due to it being used for create flow.

Is the null check necessary though? I'd expect `if (snapshot.CompositionType != CompositionType.Key)` to suffice.

Also, this if check can be combined with the enclosing if statement.

Reply:
Makes sense, the null check was just me being extra cautious.